sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ArithmeticExpressionSuite.scala

```diff
@@ -405,6 +406,28 @@ class ArithmeticExpressionSuite extends SparkFunSuite with ExpressionEvalHelper
     assert(ctx2.inlinedMutableStates.size == 1)
   }
 
+  test("SPARK-28322: IntegralDivide supports decimal type") {
+    withSQLConf(SQLConf.LEGACY_INTEGRALDIVIDE_RETURN_LONG.key -> "false") {
+      checkEvaluation(IntegralDivide(Literal(Decimal(1)), Literal(Decimal(2))), Decimal(0))
+      checkEvaluation(IntegralDivide(Literal(Decimal(1.4)), Literal(Decimal(0.6))), Decimal(2))
+      checkEvaluation(IntegralDivide(Literal(Decimal(1.2)), Literal(Decimal(1.1))), Decimal(1))
+      checkEvaluation(IntegralDivide(Literal(Decimal(1.2)), Literal(Decimal(0.0))), null)
+      checkEvaluation(DecimalPrecision.decimalAndDecimal.apply(IntegralDivide(
+        Literal(Decimal("99999999999999999999999999999999999")), Literal(Decimal(0.001)))),
```

Comment:
if we add more digits, can this overflow?

Reply:
No, we cannot overflow in this operation. Please see the code comment at https://github.com/apache/spark/pull/25136#discussion_r315016848